sklearn/cluster/affinity_propagation_.py

```diff
@@ -288,17 +295,21 @@ def fit(self, X, y=None):
         if self.affinity == "precomputed":
             self.affinity_matrix_ = X
         elif self.affinity == "euclidean":
-            self.affinity_matrix_ = -euclidean_distances(X, squared=True)
+            self.affinity_matrix_ = -euclidean_distances(X, squared=False)
```

Comment:
Add the option for a ``sqeuclidean`` as an option for affinity, but keep ``euclidean`` as default with a deprecated warning that says "in the future please use sqeuclidean for squared distances"?